libraries/chain/include/steem/chain/util/nai_generator.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <steem/protocol/asset_symbol.hpp>
+
+namespace steem { namespace chain {
+   class nai_generator {
+      /** FNV-1a implementation https://en.wikipedia.org/wiki/Fowler-Noll-Vo_hash_function#FNV-1a_hash */
+      struct hasher {
```

Comment:
Is this struct used?

Reply:
The sole purpose is to set clear boundaries between the FNV-1a implementation and the NAI logic.

Reply:
The operator overloading makes this difficult to follow.

If you use bits from the block id, it is already a hash and can be used as is. There is no reason to hash the hash. At that point, if you know the first hash, the second hash is still deterministic. There is no extra entropy being added.